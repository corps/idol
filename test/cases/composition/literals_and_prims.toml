[LitFitIntoPrim]
is_a = ["literal:string:my words", "string"]

[LitContraPrim]
is_a = ["literal:string:my words", "string"]
variance = "Contravariant"

[LitWithTags]
is_a = "literal:double:2.4"
tags = ["abc"]

[LitWithTagsContraPrim]
is_a = ["LitWithTags", "double"]
variance = "Contravariant"
