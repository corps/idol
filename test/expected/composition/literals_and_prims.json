{
  "LitContraPrim": {
    "enum": [],
    "fields": null,
    "is_a": [
      "string"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  },
  "LitFitIntoPrim": {
    "enum": [],
    "fields": null,
    "is_a": [
      "literal:string:my words"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  },
  "LitWithTags": {
    "enum": [],
    "fields": null,
    "is_a": [
      "literal:double:2.4"
    ],
    "tags": [
      "abc"
    ],
    "trim": false,
    "variance": "Covariant"
  },
  "LitWithTagsContraPrim": {
    "enum": [],
    "fields": null,
    "is_a": [
      "double"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  }
}
